test/epoch_token_locker.js

```diff
@@ -0,0 +1,216 @@
+const EpochTokenLocker = artifacts.require("EpochTokenLocker")
+const IntervalTokenStoreTestInterface = artifacts.require("IntervalTokenStoreTestInterface")
+const MockContract = artifacts.require("MockContract")
+const ERC20Interface = artifacts.require("ERC20")
+
+
+
+const truffleAssert = require("truffle-assertions")
+
+
+contract("EpochTokenLocker", async (accounts) => {
+  const [user_1] = accounts
+
+  describe("deposit", () => {
+    it("processes a deposit and stores it in the pendingDeposits", async () => {
+      const instance = await EpochTokenLocker.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(true)
+
+      await instance.deposit(ERC20.address, 100)
+      assert.equal(await instance.getPendingDepositAmount(user_1, ERC20.address), 100)
+      assert.equal(await instance.getPendingDepositBatchNumber(user_1, ERC20.address), 0)
+    })
+  
+    it("throws, if transferFrom fails", async () => {
+      const instance = await EpochTokenLocker.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(false)
+      await truffleAssert.reverts(instance.deposit(ERC20.address, 100), "Tokentransfer for deposit was not successful")
+    })
+
+    it("adds two deposits, if they are deposited during same stateIndex", async () => {
+      const instance = await EpochTokenLocker.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(true)
+      await instance.deposit(ERC20.address, 100)
+      await instance.deposit(ERC20.address, 100)
+
+      assert.equal(await instance.getPendingDepositAmount(user_1, ERC20.address), 200)
+      assert.equal(await instance.getPendingDepositBatchNumber(user_1, ERC20.address), 0)
+    })
+
+    it("does not add two deposits, if they are not deposited during same stateIndex", async () => {
```

Comment:
Wouldn't make more sense to test:

"A deposit, consolidates any pending deposit"

And ensure that the user has:
* balance: 100
* pending deposit: 100

Also, instead of using the same amount 100 + 100, consider to use 100 and 50 for example, to make sure they are saved in the right place